src/utils/LogsUtil.ts

```diff
@@ -160,62 +160,80 @@ export function getDayNumberFromDate(dateUnix: number): number {
     return -1;
 }
 
+/**
+ * Compares server logs with local logs and then syncs them
+ * @param logs - logs returned by the server
+ */
 export async function compareWithLocalLogs(logs: Array<Log>) {
     let localLogs: Array<Log> = getAllLogObjects();
     let changed = false;
 
+    // if local logs has more entries than logs, merge them
     if (localLogs.length > logs.length) {
         return mergeLocalLogs(localLogs, logs);
     }
 
-    for (let i = 0; i < localLogs.length; i++) {
-        if (
-            logs[i].day_number !== localLogs[i].day_number ||
-            !compareDates(new Date(logs[i].date), new Date(localLogs[i].date))
-        ) {
+    // if local logs have fewer entries than logs, add server logs to local
+    if (localLogs.length < logs.length) {
+        logs.forEach(log => {
+            // check if a localLog for that day number exists
+            const foundDay = localLogs.find(e => e.day_number === log.day_number);
```

Comment:
It will not work in cases where a local log is already generated before compare logs. That is why we also compare it with dates.